packages/node-patches/BUILD.bazel

```diff
@@ -55,14 +55,12 @@ tsc(
 # Like the generated mocha_test but we don't want to run the patches before testing the patches
 nodejs_test(
     name = "unit_test",
-    args = ["$(location %s)" % s for s in test_js],
     data = js + test_js + [
         "@npm_node_patches//:node_modules",
     ],
     entry_point = "@npm_node_patches//:node_modules/mocha/bin/mocha",
-    tags = [
-        "fix-windows",
-    ],
+    tags = ["fix-windows"],
```

Comment:
shouldn't this work on windows now?

Reply:
I think it ran into some other error. I'll enable it and comment on the error.

Reply:
Yep. There are still failures on Windows in the node-patches test once enabled:

/cc @soldair 

https://buildkite.com/bazel/rules-nodejs-nodejs/builds/5480#bda746b5-f98d-4fc0-a4f5-9bb04c0a254e:
```
==================== Test output for //packages/node-patches:unit_test:
--
  | ["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe",["--preserve-symlinks","--preserve-symlinks-main"],["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe","npm_node_patches/node_modules/mocha/bin/mocha","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/escape.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/lstat.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/opendir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readdir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readlink.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/realpath.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/shell-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/spawn-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/test.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/worker-threads-script.js"],"D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\Windows;C:\\Windows\\System32;C:\\Windows\\System32\\WindowsPowerShell\\v1.0"]
  | ["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe",["--preserve-symlinks","--preserve-symlinks-main"],["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe","npm_node_patches/node_modules/mocha/bin/mocha","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/escape.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/lstat.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/opendir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readdir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readlink.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/realpath.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/shell-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/spawn-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/test.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/worker-threads-script.js"],"D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\Windows;C:\\Windows\\System32;C:\\Windows\\System32\\WindowsPowerShell\\v1.0"]
  |  
  |  
  | escape function
  | 1) isOutPath is correct
  | 2) isEscape is correct
  | √ isEscape handles relative paths
  |  
  | testing lstat
  | {"mainThread":["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe",["--preserve-symlinks","--preserve-symlinks-main"],["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe","npm_node_patches/node_modules/mocha/bin/mocha","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/escape.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/lstat.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/opendir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readdir.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/readlink.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/fs/realpath.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/shell-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/spawn-script.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/test.js","D:/b/ozpmhnle/execroot/build_bazel_rules_nodejs/bazel-out/x64_windows-fastbuild/bin/packages/node-patches/test/subprocess/worker-threads-script.js"],"D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\Windows;C:\\Windows\\System32;C:\\Windows\\System32\\WindowsPowerShell\\v1.0"],"worker":["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe",["--preserve-symlinks","--preserve-symlinks-main"],["D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs\\node.exe","D:\\b\\ozpmhnle\\execroot\\build_bazel_rules_nodejs\\bazel-out\\x64_windows-fastbuild\\bin\\packages\\node-patches\\test\\subprocess\\worker-threads-script.js"],"D:\\b\\ozpmhnle\\external\\nodejs_windows_amd64\\bin\\nodejs;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\tools\\msys64\\usr\\bin;C:\\Windows;C:\\Windows\\System32;C:\\Windows\\System32\\WindowsPowerShell\\v1.0"]}
  | √ can lstat symlink in root
  | √ lstat of symlink out of root is file.
  | √ not in root, symlinks do what they would have before.
  |  
  | testing opendir
  | 1 opendir: found link?  D:\temp\tmp-10368Zd1wSD3V59fA\a\apples false
  | 2 opendir: found link?  D:\temp\tmp-10368Zd1wSD3V59fA\a\link true
  | 2 opendir: escapes? [target] D:\temp\tmp-10368Zd1wSD3V59fA\b\file [link] D:\temp\tmp-10368Zd1wSD3V59fA\a\link false D:\temp\tmp-10368Zd1wSD3V59fA
  | 3) can opendir dirent in root
  | 1 opendir: found link?  D:\temp\tmp-10368DO9WPRyRX4cQ\a\apples false
  | 2 opendir: found link?  D:\temp\tmp-10368DO9WPRyRX4cQ\a\link true
  | 2 opendir: escapes? [target] D:\temp\tmp-10368DO9WPRyRX4cQ\b\file [link] D:\temp\tmp-10368DO9WPRyRX4cQ\a\link true D:\temp\tmp-10368DO9WPRyRX4cQ\a
  | 2 opendir: patching dirent to look like it's target D:\temp\tmp-10368DO9WPRyRX4cQ\b\file
  | Dirent { name: 'apples', [Symbol(type)]: 1 } Dirent {
  | name: 'link',
  | isFile: [Function],
  | isDirectory: [Function],
  | isBlockDevice: [Function],
  | isCharacterDevice: [Function],
  | isSymbolicLink: [Function],
  | isFIFO: [Function],
  | isSocket: [Function],
  | [Symbol(type)]: 3
  | }
  | 4) can opendir dirent link out of root
  | 1 opendir: found link?  D:\temp\tmp-10368T0ZXzrRSlk0r\a\apples false
  | 2 opendir: found link?  D:\temp\tmp-10368T0ZXzrRSlk0r\a\link true
  | 2 opendir: escapes? [target] D:\temp\tmp-10368T0ZXzrRSlk0r\b\file [link] D:\temp\tmp-10368T0ZXzrRSlk0r\a\link false D:\temp\tmp-10368T0ZXzrRSlk0r
  | √ can async iterate opendir
  | 1 opendir: found link?  D:\temp\tmp-10368Qoz16IUl0FJ1\a\apples false
  | 2 opendir: found link?  D:\temp\tmp-10368Qoz16IUl0FJ1\a\link true
  | 2 opendir: escapes? [target] D:\temp\tmp-10368Qoz16IUl0FJ1\b\file [link] D:\temp\tmp-10368Qoz16IUl0FJ1\a\link true D:\temp\tmp-10368Qoz16IUl0FJ1\a
  | 2 opendir: patching dirent to look like it's target D:\temp\tmp-10368Qoz16IUl0FJ1\b\file
  | √ can async iterate opendir link out of root
  |  
  | testing readdir
  | √ can readdir dirent in root
  | √ can readdir link dirents as files out of root
  |  
  | testing readlink
  | √ can resolve symlink in root
  | √ doesn't resolve as symlink outside of root
  |  
  | testing realpath
  | √ can resolve symlink in root
  | √ doesn't resolve as symlink outside of root
  |  
  | spawning child processes
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  | 5) get patched if run as shell script.
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  | 6) overwrites spawn related variables correctly.
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  | 7) can spawn node from the shell
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  | 8) can spawn node from spawn
  |  
  |  
  | 12 passing (167ms)
  | 8 failing
  |  
  | 1) escape function
  | isOutPath is correct:
  |  
  | AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  |  
  | assert.ok(!isOutPath('/a/b'))
  |  
  | + expected - actual
  |  
  | -false
  | +true
  |  
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\fs\escape.js:27:16)
  | at processImmediate (internal/timers.js:439:21)
  |  
  | 2) escape function
  | isEscape is correct:
  |  
  | AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
  |  
  | assert.ok(isEscape('/a/c/boop', '/a/b/l'))
  |  
  | + expected - actual
  |  
  | -false
  | +true
  |  
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\fs\escape.js:33:16)
  | at processImmediate (internal/timers.js:439:21)
  |  
  | 3) testing opendir
  | can opendir dirent in root:
  | Error: EPERM: operation not permitted, lstat 'D:\temp\tmp-10368Zd1wSD3V59fA\a'
  | at Object.lstatSync (fs.js:906:3)
  | at _rmdirRecursiveSync (D:\b\ozpmhnle\external\npm_node_patches\node_modules\tmp\lib\tmp.js:300:19)
  | at Object._cleanupCallback [as removeCallback] (D:\b\ozpmhnle\external\npm_node_patches\node_modules\tmp\lib\tmp.js:446:7)
  | at Object.withFixtures (D:\b\ozpmhnle\external\npm_node_patches\node_modules\inline-fixtures\build\src\fixtures.js:72:17)
  | at async Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\fs\opendir.js:35:9)
  |  
  | 4) testing opendir
  | can opendir dirent link out of root:
  | Error: EPERM: operation not permitted, lstat 'D:\temp\tmp-10368DO9WPRyRX4cQ\a'
  | at Object.lstatSync (fs.js:906:3)
  | at _rmdirRecursiveSync (D:\b\ozpmhnle\external\npm_node_patches\node_modules\tmp\lib\tmp.js:300:19)
  | at Object._cleanupCallback [as removeCallback] (D:\b\ozpmhnle\external\npm_node_patches\node_modules\tmp\lib\tmp.js:446:7)
  | at Object.withFixtures (D:\b\ozpmhnle\external\npm_node_patches\node_modules\inline-fixtures\build\src\fixtures.js:72:17)
  | at async Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\fs\opendir.js:61:9)
  |  
  | 5) spawning child processes
  | get patched if run as shell script.:
  | Error: Command failed: NP_SUBPROCESS_BIN_DIR=D:\temp\node-patches-test-tmp node -r D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\register.js -e 'console.log(JSON.stringify([process.execPath,process.execArgv,process.argv,process.env.PATH]))'
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  |  
  | at checkExecSyncError (child_process.js:621:11)
  | at Object.execSync (child_process.js:657:15)
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\test.js:34:24)
  | at processImmediate (internal/timers.js:439:21)
  |  
  | 6) spawning child processes
  | overwrites spawn related variables correctly.:
  | Error: Command failed: NP_SUBPROCESS_BIN_DIR=D:\temp\node-patches-test-tmp node -r D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\register.js D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\worker-threads-script.js
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  |  
  | at checkExecSyncError (child_process.js:621:11)
  | at Object.execSync (child_process.js:657:15)
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\test.js:38:24)
  | at processImmediate (internal/timers.js:439:21)
  |  
  | 7) spawning child processes
  | can spawn node from the shell:
  | Error: Command failed: NP_SUBPROCESS_BIN_DIR=D:\temp\node-patches-test-tmp node -r D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\register.js D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\shell-script.js
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  |  
  | at checkExecSyncError (child_process.js:621:11)
  | at Object.execSync (child_process.js:657:15)
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\test.js:44:24)
  | at processImmediate (internal/timers.js:439:21)
  |  
  | 8) spawning child processes
  | can spawn node from spawn:
  | Error: Command failed: NP_SUBPROCESS_BIN_DIR=D:\temp\node-patches-test-tmp node -r D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\register.js D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\spawn-script.js
  | 'NP_SUBPROCESS_BIN_DIR' is not recognized as an internal or external command,
  | operable program or batch file.
  |  
  | at checkExecSyncError (child_process.js:621:11)
  | at Object.execSync (child_process.js:657:15)
  | at Context.<anonymous> (D:\b\ozpmhnle\execroot\build_bazel_rules_nodejs\bazel-out\x64_windows-fastbuild\bin\packages\node-patches\test\subprocess\test.js:52:24)
  | at processImmediate (internal/timers.js:439:21)
  |  
  |  
  |  
  | ================================================================================


```